Maple2.Server.Game/Manager/Config/BuffManager.cs

```diff
@@ -0,0 +1,234 @@
+            buff.Disable();
+        }
+
+        logger.Information("{Id} AddBuff to {ObjectId}: {SkillId},{Level} for {Tick}ms", buff.ObjectId, owner.ObjectId, id, level, buff.EndTick - buff.StartTick);
+        // Logger.Information("> {Data}", additionalEffect.Property);
+        if (notifyField) {
+            owner.Field.Broadcast(BuffPacket.Add(buff));
+        }
+    }
+
+    public virtual void Update(long tickCount) {
+        foreach (Buff buff in Buffs.Values) {
+            buff.Activate();
+
+            if (buff.Expired(tickCount)) {
+                Remove(buff.Id);
+                continue;
+            }
+
+            // TODO: Check conditions less frequently
+            if (!buff.UpdateEnabled()) {
+                continue;
+            }
+
+            // TODO: Move this to a better location?
+            if (actor.Field.Metadata.Property.Type == MapType.Pvp && buff.Metadata.Property.RemoveOnPvpZone) {
+                Remove(buff.Id);
+                continue;
+            }
+
+            buff.Proc(tickCount);
+        }
+    }
+
+    public void LeaveField() {
+        foreach (MapEntranceBuff buff in actor.Field.Metadata.EntranceBuffs) {
+            Remove(buff.Id);
+        }
+        foreach (Buff buff in Buffs.Values) {
+            if (buff.Metadata.Property.RemoveOnLeaveField) {
+                Remove(buff.Id);
+            }
+        }
+    }
+
+    private void AddMapBuffs() {
+        foreach (MapEntranceBuff buff in actor.Field.Metadata.EntranceBuffs) {
+            AddBuff(actor, actor, buff.Id, buff.Level);
+        }
+
+        if (actor.Field.Metadata.Property.Type == MapType.Pvp) {
+            foreach (Buff buff in Buffs.Values) {
+                if (buff.Metadata.Property.RemoveOnPvpZone) {
+                    Remove(buff.Id);
+                }
+
+                if (!buff.Metadata.Property.KeepOnEnterPvpZone) {
+                    Remove(buff.Id);
+                }
+            }
+        }
+
+        if (actor.Field.Metadata.Property.Region == MapRegion.ShadowWorld) {
+            AddBuff(actor, actor, Constant.shadowWorldBuffHpUp, 1);
+            AddBuff(actor, actor, Constant.shadowWorldBuffMoveProtect, 1);
+        }
+    }
+
+    public void AddItemBuffs(Item item) {
+        foreach (ItemMetadataAdditionalEffect buff in item.Metadata.AdditionalEffects) {
+            AddBuff(actor, actor, buff.Id, buff.Level);
+        }
+    }
+
+    public void RemoveItemBuffs(Item item) {
+        foreach (ItemMetadataAdditionalEffect buff in item.Metadata.AdditionalEffects) {
+            Remove(buff.Id);
+        }
+    }
+
+    public bool Remove(int id) {
+        if (!Buffs.TryGetValue(id, out Buff? buff)) {
+            return false;
+        }
+
+        //TODO: Check if buff is removable/should be removed
+        Buffs.Remove(id, out _);
+        actor.Field.Broadcast(BuffPacket.Remove(buff));
+        return true;
+
+    }
+
+    public void RemoveAll() {
```

Comment:
Do any packets need to be sent here?

Reply:
I meant whether or not we need to send the packet to remove the buffs instead of just clearing them. Also this function doesn't look like a strict "RemoveAll" anymore.

Reply:
oh! no we don't need to send packets for it. I was mainly considering using this for the job change command but yeah that didn't make sense. I removed it and put the proper calls on the command instead.

Reply:
btw `Buffs` is public so if you're just going to use this for a command you can probably just remove this function entirely and avoid any confusion.